alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/task/StartupScheduledTask.java

```diff
@@ -31,18 +31,26 @@
 
     private final TaskManager taskManager;
 
-    public StartupScheduledTask(final TaskScheduler taskScheduler, final String taskName, final TaskManager taskManager) {
+    public StartupScheduledTask(TaskScheduler taskScheduler, String taskName, TaskManager taskManager) {
         super(taskScheduler, taskName);
         this.taskManager = taskManager;
     }
 
     public abstract String scheduleCronExpression();
 
+    public boolean shouldScheduleTask() {
+        return true;
+    }
+
     public void startTask() {
+        boolean shouldScheduleTask = shouldScheduleTask();
+        if (!shouldScheduleTask) {
```

Comment:
Can we put a log in this if statement to specify why the task didn't run? Perhaps Debug?